### PR TITLE
Fix crash possibility if `module: <name>` is not defined and on mistake `position: <position>`

### LINKED ISSUE #3442: Crash on Start possibility
I just see crash possibility:
* if `module:` is missing in module configuration object
* module configuration object is empty

crash return:
```
[2024-05-07 12:54:10.972] [ERROR] (node:18916) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'split')
    at loadModule (/home/bugsounet/MagicMirror/js/app.js:156:27)
    at loadModules (/home/bugsounet/MagicMirror/js/app.js:213:10)
    at async App.start (/home/bugsounet/MagicMirror/js/app.js:261:3)
(Use `electron --trace-warnings ...` to show where the warning was created) 
```

I will see to patch it

### PR DESCRIPTION
**Pull request by bugsounet**
Fix #3442

### CODE REVIEW

**Comment by KristjanESPERANTO** [comment]
Just a thought: Wouldn't that be a thing that belongs in `check_config.js`?

**Comment by bugsounet** [comment]
`check_config.js` check only syntax but it does not check the contents of the file

should we make some other checks on it too ?

**Comment by sdetweil** [comment]
well, this one is the first actual content that is important.. 
altho position  should be checked too.. at least valid if specified.. that is another crash..

**Comment by bugsounet** [comment]
So, we have to check `module: <name>`and `position: <valid position>` (if defined)

we check it on `check_config.js` and MM core ?

**Comment by sdetweil** [comment]
I think so.. never crash..

**Comment by bugsounet** [comment]
ok, I will see to code it

**Comment by bugsounet** [comment]
Right, I will try to add needed rules with eslint in `check_config.js` ;)

I have never done this kind of test.
bugsounet, it's time to learn :)

**Comment by bugsounet** [comment]
module name error:
image

postion error:
image

No error:
image

I'm not sure that i can integrate it into eslint

**Comment by bugsounet** [comment]
Before approuve some new deps.
Thanks to review this :>

**Comment by khassel** [comment]
If I use a module with a valid position and `disabled: true` , e.g.

```js
		{
			module: "weather",
			header: "default weather module current",
			position: "top_left",
			disabled: true,
			config: {
				weatherProvider: "openmeteo",
				lat: 50.17659,
				lon: 8.62685,
				useCorsProxy: true,
				roundTemp: true,
				showPrecipitationProbability: true,
				showHumidity: "wind",
				showUVIndex: true,
				appendLocationNameToHeader: false,
				windUnits: "kmh",
				showFeelsLike: false,
				bridgeId: "abc",
			},
		},
```

I get a warning

```bash
[2024-06-07 23:21:00.091] [WARN]  No module name found for this configuration: {
  module: 'weather',
  header: 'default weather module current',
  position: 'top_left',
  disabled: true,
  config: {
    weatherProvider: 'openmeteo',
    lat: 50.17659,
    lon: 8.62685,
    useCorsProxy: true,
    roundTemp: true,
    showPrecipitationProbability: true,
    showHumidity: 'wind',
    showUVIndex: true,
    appendLocationNameToHeader: false,
    windUnits: 'kmh',
    showFeelsLike: false,
    bridgeId: 'abc'
  }
}
```

Is this intended? I think we need no such warning for disabled modules ...

**Comment by sdetweil** [comment]
i agree, no warning there.

**Comment by bugsounet** [comment]
sure, it's not logical

**Comment by bugsounet** [comment]
fixed

**Review by khassel — APPROVED** [review]

File: js/loader.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const Loader = (function () {
 	const loadedModuleFiles = [];
 	const loadedFiles = [];
 	const moduleObjects = [];
+	const positions = ["top_bar", "top_left", "top_center", "top_right", "upper_third", "middle_center", "lower_third", "bottom_left", "bottom_center", "bottom_right", "bottom_bar", "fullscreen_above", "fullscreen_below"];
```
Comment (rejas) [inline js/loader.js:10]: this code seems double, can we put it in a common class?

File: js/loader.js [hunk 2]
```diff
@@ -50,7 +51,8 @@ const Loader = (function () {
 	 * @returns {object[]} module data as configured in config
 	 */
 	const getAllModules = function () {
-		return config.modules;
+		const AllModules = config.modules.filter((module) => (module.module !== undefined) && (positions.indexOf(module.position) > -1 || typeof (module.position) === "undefined"));
```
Comment (rejas) [inline js/loader.js:54]: part of this code seems double, can we put it in a common function?

**Review by rejas — CHANGES_REQUESTED** [review]

**Comment by bugsounet** [comment]
image

It's boring to redo the packages :/

**Comment by bugsounet** [comment]
ping @rejas

File: js/loader.js [hunk 3]
```diff
@@ -50,7 +50,9 @@ const Loader = (function () {
 	 * @returns {object[]} module data as configured in config
 	 */
 	const getAllModules = function () {
-		return config.modules;
+		const positions = ["top_bar", "top_left", "top_center", "top_right", "upper_third", "middle_center", "lower_third", "bottom_left", "bottom_center", "bottom_right", "bottom_bar", "fullscreen_above", "fullscreen_below"];
```
Comment (rejas) [inline js/loader.js:53]: cant we put these positions (used also in app.js) in a common file)?

**Review by rejas — COMMENTED** [review]
ping @bugsounet

**Comment by bugsounet** [comment]
@rejas: 

* I see that postions is again used in main.js. So I use it with MM.getAvailableModulePositions
 --> I able to have all position in loader.js

* I have move *position* utils in utils.js
 --> used in app.js with Utils.moduleHasValidPosition()

**Comment by bugsounet** [comment]
@rejas: tell me

 * If it's ok for you or not ?
 * If you want really push it for MM² v2.28 ?
 
Sorry to say this but it's discouraging (this PR is open since May 10)  :/

**Comment by rejas** [comment]
> @rejas: tell me
> 
>     * If it's ok for you or not ?
> 
>     * If you want really push it for MM² v2.28 ?
> 
> 
> Sorry to say this but it's discouraging (this PR is open since May 10) :/

Of course its ok for me (I do trust your code :-) I was jsut pondering a lot if we could move ALL the position usages into the same file. but that would be as we say in germany: in schönheit sterben (meaning "to die in beauty" aka to work too much and never finish :-D

So sorry for my slow approbval, but will do now!

**Review by rejas — APPROVED** [review]

**Pull request merged**